src/odemis/driver/autoscript_client.py

```diff
@@ -1519,7 +1534,7 @@ def _acquire(self) -> None:
                     # Retrieve the image (scans image, blocks until the image is received)
                     # TODO: use the metadata from the image acquisition _md once it's available
                     image, _md = self.parent.acquire_image(self._scanner.channel)
-
+                    image = ndimage.median_filter(image, 3)  # median filter to remove noise
```

Comment:
This shouldn't be hard-coded in the driver. The driver should just pass the image as-is. Please move this to the acq/gui code, where de-noising is required.

Reply:
"""[tepals](https://github.com/tepals) [3 days ago](https://github.com/delmic/odemis/pull/3001#discussion_r1928682232)
Why do you want to remove noise? I would expect the acquire method to just acquire and not do any image adjustments.
"""

@tepals @pieleric responding to the same comment for clarity. 
Every cryo image needs to be median filtered because the data is bad, so i think the most natural place is in the driver. I've adjusted it so that it's a option in the metadata, so it's not hardcoded but configurable. 
If not here, where do you suggest? I'm not really sure of a suitable place

Reply:
How would a user normally do this? Will they select median filtering by default in the XT UI?

Reply:
not sure if you can change it in the ui, but it would be expected